### PR TITLE
Add functions to set pick and place goal fields

### PR DESCRIPTION
**Pull request by mvieth**
### Description

Add functions to set fields in the pick and place goals, namely `place_eef` and `allow_gripper_support_collision`. Also some minor formatting changes by clang-format.
See also issue #631

### CODE REVIEW

File: moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h [hunk 1]
```diff
@@ -311,6 +311,14 @@ class MoveGroupInterface
    * objects are allowed to touch the support surface */
   void setSupportSurfaceName(const std::string& name);
 
+  /** \brief For place operations, either the location of the object can be specified ( \e place_eef = false,
+   * this is the default ) or the location of the end effector ( \e place_eef = true ) */
+  void setPlaceEEF(const bool place_eef);
```
Comment (BryceStevenWilley) [inline moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h]: No need to declare parameters const like this.

File: moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h [hunk 2]
```diff
@@ -311,6 +311,14 @@ class MoveGroupInterface
...
+
+  /** \brief For pick/place operations, specify whether collisions between the gripper and the support surface should
+   * be acceptable. By default, this is true */
+  void setAllowGripperSupportCollision(const bool allow_gripper_support_collision);
```
Comment (BryceStevenWilley) [inline moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h]: Same as above.

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 3]
```diff
@@ -524,6 +526,19 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     support_surface_ = support_surface;
   }
 
+  void setPlaceEEF(const bool place_eef)
+  {
+    place_eef_ = place_eef;
+    ROS_INFO_NAMED("move_group_interface", "Place EEF: %s", place_eef_ ? "yes" : "no");
```
Comment (BryceStevenWilley) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: Not many of the other functions have logs like this. I'm fine with it, but it should be changed to debug, it'll probably just add to the noise most of the time.

**Comment by rhaschke** [comment]
Having thought a little bit more about this PR, I suggest to remove the extra member variables and instead pass those flags as arguments to the corresponding `constructGoal()` and `pick()` / `place()` functions - providing the old hard-coded values as defaults. My reasoning for this is, that those flags usually will change from call to call. What do you think @BryceStevenWilley?

**Comment by BryceStevenWilley** [comment]
I don't think those parameters should change every call. Whatever is providing the goals targets won't change it's transform math, so Place EEF shouldn't change. And, at least for every pick or every place, collision with support should also stay the same. But in general, I agree that less saved state in the class is better, so I'm okay with that implementation as well.

**Comment by mvieth** [comment]
Maybe it would even make sense to expose the `constructGoal()` functions for the Pickup and Place Goals? There are more fields and options a user might want access to (e.g. `attached_object_touch_links`, `minimize_object_distance`, `allowed_touch_objects`), and making setter functions/optional arguments for all of them would not be so elegant. Instead, the goals could be optional arguments for the pick/place functions, so that a user could call the `constructGoal()`function, then change any option in the goal, and pass it on to pick/place.

**Comment by rhaschke** [comment]
> Maybe it would even make sense to expose the `constructGoal()`

I like this idea. Maybe the function names should become more descriptive if exposed, i.e. `constructPickGoal` and `constructPlaceGoal`.

**Comment by mvieth** [comment]
Ok, I pushed the proposed changes. Looking forward to your feedback @rhaschke @BryceStevenWilley . I did not remove the `setPlaceEEF()` and `setAllowGripperTouchSupport()` functions yet because they do not conflict with the new `pick()` and `place()` functions, but could do so.

**Comment by rhaschke** [comment]
> I did not remove the setPlaceEEF() and setAllowGripperTouchSupport()

My original intention was to _get rid off_ the extra state variables :wink:
I pushed some extra commits to your PR branch to reflect my vision of this PR.
You can now decide whether to add those `place_eef`, `allow_touching_support` arguments to `constructPickGoal` and `constructPlaceGoal`.

**Comment by mvieth** [comment]
I would change [this line](https://github.com/ros-planning/moveit/pull/1498/commits/b666b9864cf5ebecd5634777c4ddb398326d4a66#diff-91269c2c54a4fe24e4a4df9d6e8db959L1086) and the previous one to how they originally were, with `goal.allow_gripper_support_collision = true` (same for `constructPlaceGoal`).

> You can now decide whether to add those place_eef, allow_touching_support arguments to constructPickGoal and constructPlaceGoal

I would say that's not necessary any more because a user could just use `constructPickupGoal` and change the `place_eef` and `allow_gripper_support_collision` fields in the goal before calling `pick(PickupGoal)`

**Comment by rhaschke** [comment]
> I would change this line and the previous one to how they originally were, with `goal.allow_gripper_support_collision = true`

I'm fine with this. Didn't noticed the original state.

**Comment by v4hn** [comment]
Thanks Robert, I wasn't sure if you are ok with squashing your commits too.

**Pull request merged**